hibernate-reactive-core/src/test/java/org/hibernate/reactive/BaseReactiveTest.java

```diff
@@ -160,6 +170,69 @@ public void after(TestContext context) {
 		return sessionFactory.unwrap( Stage.SessionFactory.class );
 	}
 
+	private static Field accessibleField(Class<?> clazz, String name) throws Exception {
+		Field field = clazz.getDeclaredField(name);
+		field.setAccessible(true);
+		return field;
+	}
+
+	private static Object accessibleFieldGet(Class<?> clazz, String name, Object instance) throws Exception {
+		return accessibleField(clazz, name).get(instance);
+	}
+
+	protected Mutiny.Session injectDelayedConnection(Mutiny.Session session, CountDownLatch synchronizer) {
+		try {
+			Object reactiveSessionImpl = accessibleFieldGet(MutinySessionImpl.class, "delegate", session);
+			Object proxyConnection = accessibleFieldGet(ReactiveSessionImpl.class, "reactiveConnection", reactiveSessionImpl);
+			Object sqlClientPool = accessibleFieldGet(proxyConnection.getClass(), "sqlClientPool", proxyConnection);
+			Pool pool = (Pool) accessibleFieldGet(DefaultSqlClientPool.class, "pool", sqlClientPool);
+			Pool delayingPool = new Pool() {
+				@Override
+				public void getConnection(Handler<AsyncResult<SqlConnection>> handler) {
+					pool.getConnection(ar -> {
+						// maybe there's a nicer way to do this asynchronously
+						new Thread(() -> {
+							try {
+								if (!synchronizer.await(60, TimeUnit.SECONDS))
+									handler.handle(Future.failedFuture("synchronizer await failed"));
+								handler.handle(ar);
```

Comment:
I honestly can't think now of a better way to introduce delay, but I think it's wrong to have the helper thread invoke a method on the original `handler` here as that would violate the assumptions.